### DISCUSSION TITLE
How can I custom the response error message in different situations?

### DISCUSSION THREAD

**Discussion by knota-lab** [rwf2/Rocket#1822] [Questions] [answered]
I want to use this err as the response content.
![image](https://user-images.githubusercontent.com/26835087/129432296-97359853-8123-4a7f-82da-b49a4bb306c3.png)

**Comment by jebrosen** [comment, marked as answer]
This is really close.

First, you'll have to change the return type of `delete_document` so you can return [either a success or error response](https://rocket.rs/v0.5-rc/guide/responses/#result):

```rust
pub async fn delete_document(conn: MysqlDBConn, ids: Json<Vec<i32>>) -> Result<String, String> { ... }
```

Then, the handler can return a response according to the result of that database operation:

```rust
let res = conn.run(...).await;
match res {
    Ok(count) => Ok(format!("Deleted {} records", count)),
    Err(e) => Err(e.to_string()),
}
```

And, you can change `Result<String, String>` to whatever you like depending on how that route needs to respond - e.g. `Result<Template, Json<MadeUpErrorType>>`.
